src/plugins/kibana_overview/public/components/overview/overview.test.mocks.tsx

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+
+export const hasUserDataViewMock = jest.fn();
+
+jest.doMock('../../../../../../src/plugins/kibana_react/public', () => ({
+  useKibana: jest.fn().mockReturnValue({
+    services: {
+      http: { basePath: { prepend: jest.fn((path: string) => (path ? path : 'path')) } },
+      data: {
+        indexPatterns: {
+          hasUserDataView: hasUserDataViewMock,
+        },
+      },
+      share: { url: { locators: { get: () => ({ useUrl: () => '' }) } } },
+      uiSettings: { get: jest.fn() },
+      docLinks: {
+        links: {
+          kibana: 'kibana_docs_url',
+        },
+      },
+    },
+  }),
+  RedirectAppLinks: jest.fn((element: JSX.Element) => element),
+  overviewPageActions: jest.fn().mockReturnValue([]),
+  OverviewPageFooter: jest.fn().mockReturnValue(<></>),
```

Comment:
optional: IMO `React.createElement(React.Fragment..)` is more explicit in this case (and allows removing `*.tsx` extension)